addons/account/migrations/12.0.1.1/openupgrade_analysis_work.txt

```diff
@@ -0,0 +1,311 @@
+
+account      / account.bank.statement.line / account_number (char)         : NEW
+# DONE: post-migration: filled using res.partner.bank acc_number
+
+account      / account.chart.template   / bank_account_code_prefix (char): now required
+account      / account.chart.template   / cash_account_code_prefix (char): now required
+account      / account.chart.template   / transfer_account_code_prefix (char): NEW required: required
+account      / account.chart.template   / transfer_account_id (many2one): DEL relation: account.account.template, required: required
+account      / account.chart.template   / company_id (many2one)         : DEL relation: res.company
+# DONE: post-migration: tried to fill all of them them using the company
+
+account      / account.fiscal.year      / company_id (many2one)         : NEW relation: res.company, required: required, req_default: function
+account      / account.fiscal.year      / date_from (date)              : NEW required: required
+account      / account.fiscal.year      / date_to (date)                : NEW required: required
+account      / account.fiscal.year      / name (char)                   : NEW required: required
+# NOTHING TO DO (new model)
+
+account      / account.group            / parent_left (integer)         : DEL
+account      / account.group            / parent_path (char)            : NEW 
+account      / account.group            / parent_right (integer)        : DEL 
+# DONE: post-migration: call _parent_store_compute() method
+
+account      / account.incoterms        / active (boolean)              : NEW
+account      / account.incoterms        / code (char)                   : NEW required: required
+account      / account.incoterms        / name (char)                   : NEW required: required
+# NOTHING TO DO (rename done in apriori)
+
+account      / account.invoice          / activity_date_deadline (date) : now a function
+account      / account.invoice          / incoterm_id (many2one)        : previously in module l10n_be_intrastat
+account      / account.invoice          / message_last_post (datetime)  : DEL 
+account      / account.invoice          / message_main_attachment_id (many2one): NEW relation: ir.attachment
+# NOTHING TO DO
+
+account      / account.invoice          / reference_type (selection)    : DEL required: required, selection_keys: function, req_default: function
+account      / account.invoice          / source_email (char)           : NEW
+account      / account.invoice          / state (selection)             : selection_keys is now '['cancel', 'draft', 'in_payment', 'open', 'paid']' ('['cancel', 'draft', 'open', 'paid']')
+account      / account.invoice          / vendor_bill_id (many2one)     : NEW relation: account.invoice
+# NOTHING TO DO
+
+account      / account.invoice.line     / display_type (selection)      : NEW selection_keys: ['line_note', 'line_section']
+# DONE: post-migration: filled as 'line_section' from layout_category_id
```

Comment:
What you have to do is to:

* Sort lines by old `layout_category_id`. I think this was previously done on v11 through model order, but now it should be by sequence.
* Create one new line before first line of each section, with the `display_type` = `line_section`, and `name` like section name.

As this has to be done the same on `sale` module, maybe you can refactor somehow the method for not duplicating code.

Reply:
It's sad there is not an openupgrade method to reorder table records safely 😔

Reply:
Well, that's have not been a need till now, and I don't think we should order permanently by hard ID, but use sequence field values.